bciers/libs/shared/components/src/lib/form/FormBase.tsx

```diff
@@ -0,0 +1,162 @@
+  statutory_declaration: "Attachment is mandatory.",
+  "postal-code": "Format should be A1A 1A1",
+  phone: "Format should be ### ### ####",
+  email: "Please enter a valid email address, e.g. mail@example.com",
+  uri: "Please enter a valid website link, e.g. http://www.website.com, https://www.website.com",
+};
+
+interface Error {
+  message: string;
+  name: string;
+  params?: { [key: string]: string };
+  property: string;
+  schemaPath?: string;
+  stack: string;
+}
+
+const transformErrors = (errors: RJSFValidationError[]) => {
+  return customTransformErrors(errors, customFormatsErrorMessages);
+};
+
+const validator = customizeValidator({ customFormats });
+
+interface FormPropsWithTheme<T> extends Omit<FormProps<T>, "validator"> {
+  children?: React.ReactNode | true;
+  onChange?: (e: IChangeEvent) => void;
+  onLiveValidation?: (isValid: boolean) => void;
+  theme?: ThemeProps;
+  validator?: any;
+  setErrorReset?: (error: undefined) => void;
+  triggerValidation?: boolean;
+}
+
+const FormBase: React.FC<FormPropsWithTheme<any>> = (props) => {
+  const {
+    children,
+    disabled,
+    formData,
+    omitExtraData,
+    onChange,
+    onSubmit,
+    onLiveValidation,
+    readonly,
+    setErrorReset,
+    theme,
+    triggerValidation,
+  } = props;
+  const formTheme = disabled || readonly ? readOnlyTheme : defaultTheme;
+  const Form = useMemo(() => withTheme(theme ?? formTheme), [theme, formTheme]);
+  const [formSubmitState, setFormSubmitState] = useState(formData ?? {});
+  const [focusedField, setFocusedField] = useState<string | undefined>();
+
+  // Handling form state externally as RJSF was resetting the form data on submission and
+  // creating buggy behaviour if there was an API error and the user attempted to resubmit
+  const handleSubmit = (e: IChangeEvent) => {
+    setFormSubmitState(e.formData);
+    if (setErrorReset) setErrorReset(undefined); // Reset error state on form submission
+
+    if (onSubmit) onSubmit(e, formSubmitState);
+  };
+
+  const formRef = createRef<any>();
+
+  const validateSingleField = (field: string) => {
+    // RJSF does not support single field validation out of the box
+    // This function is a workaround to validate a single field
+    const $this = formRef.current;
+    const state = $this?.state;
+    if (!state) return;
+    const { formData, errors, errorSchema } = $this?.state;
+
+    const { errors: _errors, errorSchema: _errorSchema } =
+      $this.validate(formData);
+
+    const prevOtherFieldErrors = errors.filter(
+      (error: Error) => error.property !== `.${field}`,
+    );
+
+    const fieldErrors = _errors.filter(
+      (error: Error) => error.property === `.${field}`,
+    );
+
+    const fieldErrorSchema = _errorSchema[field];
+
+    $this.setState({
+      errors: [...new Set([...prevOtherFieldErrors, ...fieldErrors])],
+      errorSchema: { ...errorSchema, [field]: fieldErrorSchema },
+    });
+  };
```

Comment:
@shon-button this was my branch I saved a copy of and abandoned for single field validation. This is most of the relevant code, aside from the onBlur and onFocus functions, as well as updating our widgets to handle onBlur and onFocus.

The main issue with this is that it just handles flat schemas which is why it works with my previous multi-form approach.  

As soon as you start nesting objects this all falls apart. I could likely get it working with my single form approach but once we start adding complex forms and array field it will all fall apart. In the single form approach RJSF will append the section names to each field that it passes through the onBlur and onFocus. So a field called `first_name` will be `root_section1_first_name`. 

I do believe I could get this working with my mock schema you viewed in the actual PR, but as I said more complex schemas than that will be extremely hard to do single field validation on.